libraries/entities/src/DiffTraversal.cpp

```diff
@@ -123,10 +123,34 @@ void DiffTraversal::Waypoint::getNextVisibleElementDifferential(DiffTraversal::V
                     float distance = glm::distance(view.viewFrustum.getPosition(), cube.calcCenter()) + MIN_VISIBLE_DISTANCE;
                     float apparentAngle = cube.getScale() / distance;
                     if (apparentAngle > MIN_ELEMENT_APPARENT_ANGLE * view.lodScaleFactor) {
-                        if (view.viewFrustum.calculateCubeKeyholeIntersection(cube) != ViewFrustum::OUTSIDE) {
-                            next.element = nextElement;
-                            next.intersection = ViewFrustum::OUTSIDE;
-                            return;
+                        if (view.viewFrustum.cubeIntersectsKeyhole(cube)) {
+                            ViewFrustum::intersection lastIntersection = lastView.viewFrustum.calculateCubeKeyholeIntersection(cube);
+                            if (lastIntersection != ViewFrustum::INSIDE || nextElement->getLastChanged() > lastView.startTime) {
+                                next.element = nextElement;
+                                // NOTE: for differential case next.intersection is against the lastView
+                                // because this helps the "external scan" optimize its culling
+                                next.intersection = lastIntersection;
+                                return;
+                            } else {
+                                distance = glm::distance(lastView.viewFrustum.getPosition(), cube.calcCenter()) + MIN_VISIBLE_DISTANCE;
+                                apparentAngle = cube.getScale() / distance;
+                                if (!fullTraversal) {
+                                    // check for LOD truncation in the last traversal because
+                                    // we may need to traverse this element after all if the lastView skipped it for LOD
+                                    if (apparentAngle <= MIN_ELEMENT_APPARENT_ANGLE * lastView.lodScaleFactor) {
+                                        next.element = nextElement;
+                                        // element's intersection with lastView was effectively OUTSIDE
+                                        next.intersection = ViewFrustum::OUTSIDE;
+                                        return;
+                                    }
+                                } else {
+                                    // if it's a fullTraversal, we need to traverse all the children, but we may not need to scan them
+                                    // if we could see this element before, we don't need to scan it
+                                    next.element = nextElement;
+                                    next.intersection = apparentAngle > MIN_ELEMENT_APPARENT_ANGLE * lastView.lodScaleFactor ? ViewFrustum::INSIDE : ViewFrustum::OUTSIDE;
+                                    return;
+                                }
```

Comment:
This code is correct, however it took me a while to think it all the way through and reach that conclusion... and I'm fully conversant with what we're trying to do here.  I think it would help to organize/comment this code so that it will make more sense to future devs, but I don't consider that a blocker for testing this code or merging it once we're convinced it is all Good.